Source/Core/Core/HW/GCMemcard/GCMemcard.cpp

```diff
@@ -1458,6 +1458,39 @@ std::pair<u32, u32> Header::CalculateSerial() const
   return std::make_pair(serial1, serial2);
 }
 
+Header::ValidityCheckErrorCode Header::SeemsValid(u16 card_size_mbits) const
+{
+  if (m_size_mb != card_size_mbits)
+    return ValidityCheckErrorCode::INVALID_CARD_SIZE;
+
+  // only known valid values are 0 for western and 1 for shift-jis
```

Comment:
Are you sure? According to https://bugs.dolphin-emu.org/issues/10247 (see also the corresponding PR), Shift-JIS is not necessarily 1.

Reply:
All my Japanese memory cards have a 1 there, but huh, interesting. Maybe anything nonzero counts then.

Reply:
According to all the research I've done Shift-JIS isn't explicitly checked, only Western, so Shift-JIS would be implicitly non-zero.